screenpipe-js/cli/src/commands/pipe/publish.ts

```diff
@@ -306,19 +325,99 @@ export const publishCommand = new Command("publish")
+              Authorization: `Bearer ${apiKey}`,
+              "Content-Type": "application/json",
+            },
+            timeout: 30000, // 30 second timeout
+          });
+        } catch (error) {
+          // Handle version conflict specifically
+          if (axios.isAxiosError(error) && error.response) {
+            const status = error.response.status;
+            const errorData = error.response.data;
+            
+            console.log(colors.dim(`${symbols.arrow} server responded with status: ${status}`));
+            
+            // Check if this is a version conflict error
+            if (status === 400 && typeof errorData === 'string' && 
+                errorData.includes('already exists') && errorData.includes('version')) {
+              
+              // Ask user if they want to bump the version
+              const readline = require('readline').createInterface({
+                input: process.stdin,
+                output: process.stdout
+              });
+              
+              const newVersion = bumpVersion(packageJson.version);
+              
+              const question = `\n${symbols.info} ${colors.info(`Version ${packageJson.version} already exists.`)} 
+${colors.info(`Would you like to bump to version ${newVersion} and continue? (y/n): `)}`;
+              
+              const answer = await new Promise<string>(resolve => {
+                readline.question(question, (ans: string) => {
+                  readline.close();
+                  resolve(ans.toLowerCase());
+                });
+              });
+              
+              if (answer === 'y' || answer === 'yes') {
+                // Update package.json with new version
+                updatePackageVersion(newVersion);
+                logger.success(`${symbols.success} Updated package.json to version ${newVersion}`);
+                
+                // Update packageJson in memory
+                packageJson.version = newVersion;
+                
+                // Retry the request with new version
+                console.log(colors.dim(`${symbols.arrow} retrying with new version: ${newVersion}`));
+                urlResponse = await axios.post(`${API_BASE_URL}/api/plugins/publish`, {
+                  name: opts.name,
+                  version: newVersion,
+                  fileSize,
+                  fileHash,
+                  description,
+                }, {
+                  headers: {
+                    Authorization: `Bearer ${apiKey}`,
+                    "Content-Type": "application/json",
+                  },
+                  timeout: 30000,
+                });
+              } else {
+                // User chose not to bump version
+                throw new Error(`Publishing canceled. Please update the version manually in package.json.`);
+              }
+            } else {
+              // Handle other errors
+              if (typeof errorData === 'string') {
```

Comment:
@m13v this is really hard to read / debug - check early return pattern:

https://medium.com/swlh/return-early-pattern-3d18a41bba8